utils/hints_test.go

```diff
@@ -219,7 +234,8 @@ func TestGenerateHints(t *testing.T) {
 				continue
 			}
 		}
-		assert.Equal(t, test.result, GenerateHints(annMap, "foobar", "co.elastic"))
+		generateHints, _ := GenerateHints(annMap, "foobar", "co.elastic", allSupportedHints)
```

Comment:
Why not start checking the error as well in the test?

Reply:
Done in [returning list with unsupported hints and fixing test to check those](https://github.com/elastic/elastic-agent-autodiscover/pull/81/commits/93065164edbb02b7de267bfd26c1ce05cc749280)